src/main/java/io/cryostat/platform/PlatformClient.java

```diff
@@ -55,6 +57,21 @@ default void load(Promise<EnvironmentNode> promise) {
 
     List<ServiceRef> listDiscoverableServices();
 
+    default List<ServiceRef> listUniqueReachableServices() {
+        Set<String> uniqueIds = new HashSet<>();
+        return listDiscoverableServices().stream()
+                .filter((ref) -> ref.getJvmId() != null && uniqueIds.add(ref.getJvmId()))
+                .toList();
+    }
+
+    default boolean containsDuplicateTarget(ServiceRef ref) {
```

Comment:
Oh, I meant just name the method simply `contains`, ie define it as `default boolean contains(ServiceRef ref)`.